pkg/ctl/create/cluster_test.go

```diff
@@ -326,9 +338,60 @@ var _ = Describe("create cluster", func() {
 			Expect(clusterConfig.PrivateCluster.Enabled).To(BeTrue())
 			p.MockEKS().AssertNumberOfCalls(GinkgoT(), "UpdateClusterConfig", 1)
 		}
+
+		if FakeInstallerTaskCreator.CreateStub != nil {
+			Expect(FakeInstallerTaskCreator.CreateCallCount()).To(Equal(1))
+		}
 	},
 		Entry("standard cluster", createClusterEntry{}),
 
+		Entry("[Cluster with Karpenter] installs Karpenter successfully", createClusterEntry{
+			updateClusterConfig: func(c *api.ClusterConfig) {
+				c.Karpenter = &api.Karpenter{
+					Version: "v0.18.0",
+				}
+			},
+			configureKarpenterInstaller: func(ki *karpenterfakes.FakeInstallerTaskCreator) {
+				ki.CreateStub = func(ctx context.Context) error {
+					return nil
+				}
+				createKarpenterInstaller = func(ctx context.Context, cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager manager.StackManager, clientSet kubernetes.Interface, restClientGetter *kubernetes.SimpleRESTClientGetter) (karpenteractions.InstallerTaskCreator, error) {
+					return ki, nil
+				}
+			},
+		}),
+
+		Entry("[Cluster with Karpenter] fails to create the installer", createClusterEntry{
+			updateClusterConfig: func(c *api.ClusterConfig) {
+				c.Karpenter = &api.Karpenter{
+					Version: "v0.18.0",
+				}
+			},
+			configureKarpenterInstaller: func(ki *karpenterfakes.FakeInstallerTaskCreator) {
+				createKarpenterInstaller = func(ctx context.Context, cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager manager.StackManager, clientSet kubernetes.Interface, restClientGetter *kubernetes.SimpleRESTClientGetter) (karpenteractions.InstallerTaskCreator, error) {
+					return ki, fmt.Errorf("")
+				}
+			},
+			expectedErr: "failed to create installer",
+		}),
+
+		Entry("[Cluster with Karpenter] fails to actually install Karpenter", createClusterEntry{
+			updateClusterConfig: func(c *api.ClusterConfig) {
+				c.Karpenter = &api.Karpenter{
+					Version: "v0.18.0",
+				}
+			},
+			configureKarpenterInstaller: func(ki *karpenterfakes.FakeInstallerTaskCreator) {
+				ki.CreateStub = func(ctx context.Context) error {
+					return fmt.Errorf("")
```

Comment:
It's best to add a non-empty error message even if it isn't being matched in the error assertion.